### ISSUE TITLE
CA1715 (IdentifiersShouldHaveCorrectPrefix) should not fire for single letter type parameter names

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#937]
#### Repro steps

Create a method with type param names with single letter, example `Foo<K, V>`

#### Expected behavior
No CA1715 fired.
I think it is extremely common practice to have type parameter names with a single letter. We should avoid firing CA1715 for such cases.

#### Actual behavior
```
Severity	Code	Description	Project	File	Line	Suppression State
Warning	CA1715	Prefix generic type parameter name V with 'T'.	DiagnosticsTestUtilities	C:\roslyn-analyzers\src\Test\Utilities\MarkupTestFile.cs	173	Active
Warning	CA1715	Prefix generic type parameter name K with 'T'.	DiagnosticsTestUtilities	C:\roslyn-analyzers\src\Test\Utilities\MarkupTestFile.cs	173	Active
```

**Comment by mavasani** [comment]
Please remove all the suppressions for this warning ID in Analyzers.sln (not yet checked in) once this is fixed.

**Comment by bkoelman** [comment]
I'll start work on a fix.

**Comment by sharwell** [comment]
I would expect the analyzer to report a warning in this case. `K, V` is normal type parameter names for a Java developer, but not a C# developer. Prior to this change, this rule provided valuable assistance for developers coming from a Java background and interested in adhering to .NET patterns. The correct solution for the above case is to rename the generic type parameters from `K, V` to `TKey, TValue`.
